From f79e3b13c163e270427178a09a699d9110a5e18c Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 20 Apr 2018 14:19:56 +0300
Subject: [PATCH 3621/5242] MLK-18104: drm: imx: dcss: Fix brightness on some
 HDMI 1.4 ports

commit  f314226dd9daed8cebb0821a7b761ac4322f6c4d from
https://source.codeaurora.org/external/imx/linux-imx.git

Some HDR TVs have multiple HDMI ports but only one of them is HDMI 2.0
compliant, hence HDR capable. The rest are HDMI 1.4.

The HDMI 1.4 ports' supported colorimetry is only REC.709 and REC.601.
However, the supported EOTF in HDR metadata block can be BT.2084 which
should be matched with REC.2020.

This patch makes sure that BT2084 is used only if colorimetry supports
REC.2020. Otherwise, REC.709 will be used.

Additionally, change a message from info to debug. No need for it to
show up all the time.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-crtc.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-crtc.c b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
index 64b1fe0..94dcb23 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-crtc.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
@@ -176,7 +176,7 @@ void dcss_crtc_setup_opipe(struct drm_crtc *crtc, struct drm_connector *conn,
 	else
 		dcss_crtc->opipe_g = G_REC709;
 
-	if (eotf & (1 << 2))
+	if ((eotf & (1 << 2)) && dcss_crtc->opipe_g == G_REC2020)
 		dcss_crtc->opipe_nl = NL_REC2084;
 	else
 		dcss_crtc->opipe_nl = NL_REC709;
@@ -196,9 +196,9 @@ void dcss_crtc_setup_opipe(struct drm_crtc *crtc, struct drm_connector *conn,
 	else
 		dcss_crtc->opipe_pix_format = DRM_FORMAT_ARGB8888;
 
-	DRM_INFO("OPIPE_CFG: gamut = %d, nl = %d, pr = %d, pix_format = %d\n",
-		 dcss_crtc->opipe_g, dcss_crtc->opipe_nl,
-		 dcss_crtc->opipe_pr, dcss_crtc->opipe_pix_format);
+	DRM_DEBUG_KMS("OPIPE_CFG: gamut = %d, nl = %d, pr = %d, pix_fmt = %d\n",
+		      dcss_crtc->opipe_g, dcss_crtc->opipe_nl,
+		      dcss_crtc->opipe_pr, dcss_crtc->opipe_pix_format);
 }
 
 int dcss_crtc_get_opipe_cfg(struct drm_crtc *crtc,
-- 
1.7.9.5

